src/apps/sources/components/SourceForm.tsx

```diff
@@ -0,0 +1,299 @@
+                                ).map(({ value, label }) => (
+                                    <MenuItem key={value} value={value}>
+                                        {label}
+                                    </MenuItem>
+                                ))}
+                            </Field>
+                            <Typography color="error" variant="caption" component="div">
+                                <ErrorMessage name="supported_locales" component="span" />
+                            </Typography>
+                        </FormControl>
+                        <Field
+                            // required
+                            fullWidth
+                            autoComplete="off"
+                            id="custom_validation_schema"
+                            name="custom_validation_schema"
+                            label="Custom Validation Schema"
+                            margin="normal"
+                            component={TextField}
+                        />
+                        <Field
+                            // required
+                            fullWidth
+                            autoComplete="off"
+                            id="external_id"
+                            name="external_id"
+                            label="External Id"
+                            margin="normal"
+                            component={TextField}
+                        />
+                        <br />
+                        <br />
+                        {viewing ? (
+                            ""
+                        ) : (
+                            <div className={classes.submitButton}>
+                                {!status ? (
+                                    <br />
+                                ) : (
+                                    <Typography
+                                        color="textSecondary"
+                                        variant="caption"
+                                        component="span"
+                                    >
+                                        {status}
+                                    </Typography>
+                                )}
+                                {!error ? (
+                                    <br />
+                                ) : (
+                                    <Typography color="error" variant="caption" component="span">
+                                        {error}
+                                    </Typography>
+                                )}
+                                <br />
+                                <Button
+                                    variant="outlined"
+                                    color="primary"
+                                    size="medium"
+                                    type="submit"
+                                    disabled={isSubmitting}
+                                >
```

Comment:
Since we are rendering the details only, we don't need to keep the edit and submit portion of the code. Not sure we are going to implement the edit and create sources in the next sprint.

Reply:
Removed kirity